src/App.tsx

```diff
@@ -1,8 +1,60 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { GoodList } from './components/GoodList';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+   const url = 'https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json';
+
+  const downloadData = async (): Promise<Good[]> => {
+    return fetch(url)
+      .then(response => {
+        return response.json();
+      });
+  };
+
+  const allButtonHandler = () => {
+    downloadData()
+      .then(list => {
+        setGoods(list);
+      });
+  };
+
+  const fiveButtonHandler = () => {
+    downloadData()
+      .then(list => {
+        list.sort((item1: Good, item2: Good) => item1.name.localeCompare(item2.name));
+        list.splice(5);
+        setGoods(list);
+      });
+  };
+
+  const redButtonHandler = () => {
+    downloadData()
+      .then(list => {
+        const newlist = list.filter((item1: Good) => item1.color === 'red');
```

Comment:
если ты опишешь что отдает функция (ранее писал), то это не нужно будет
```suggestion
        const filteredList = list.filter((item1) => item1.color === 'red');
```